docs/src/design/decoder/constraints.md

```diff
@@ -22,7 +24,7 @@ AIR constraints for the decoder involve operations listed in the table below. Fo
 | `HALT`    | $f_{halt}$   | 4      | Stack remains unchanged. |
 | `PUSH`    | $f_{push}$   | 4      | An immediate value is pushed onto the stack. |
 
-We also use the [control flow flag](../stack/op_constraints.md#control-flow-flag) $f_{ctrl}$ exposed by the VM, which is set when any one of the above control flow operations is being executed. It has degree $4$.
+We also use the [control flow flags](../stack/op_constraints.md#control-flow-flag) $f_{ctrl}$ and $f_{ctrlb}$ exposed by the VM. $f_{ctrl}$ is set when any one of the above control flow operations is being executed. $f_{ctrlb}$ is set when a control flow operation that signifies the initialization of a new control block is being executed.  $f_{ctrl}$ and $f_{ctrlb}$ have degree 4.
```

Comment:
Would it make sense to redefine hash of all blocks as always taking 8 elements as input. Specifically:

* `JOIN`: hash(left, right)
* `SPLIT`: hash(left, right)
* `LOOP`: hash(body, 0)
* `CALL`: hash(target, 0)
* `SYSCALL`: hash(target, 0)

This way constraints for block hash computation become simpler and we may not need $f_{ctrlb}$ flag.

Reply:
After review of the code it looks like the blocks are always [padded](https://github.com/0xPolygonMiden/miden-vm/blob/next/processor/src/chiplets/mod.rs#L240).

Reply:
I've removed the $f_{ctrlb}$ flag and embedded the $\alpha_5 * opcode$ value directly in the $u_{join}$, $u_{split}$, ... terms.  I believe this will yield a lower degree.  I think the previous pattern of having the opcode calculation within $h_{init}$ wouldn't have worked as it would have yielded a higher degree for the $u_{join}$, ... terms.

Reply:
> After review of the code it looks like the blocks are always [padded](https://github.com/0xPolygonMiden/miden-vm/blob/next/processor/src/chiplets/mod.rs#L240).

Yep - seems like it. Though, I would probably use something like [this](https://github.com/0xPolygonMiden/miden-vm/blob/next/core/src/program/blocks/loop_block.rs#L31) as the "source of truth".